src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -283,6 +289,13 @@ void CoreWorkerDirectTaskReceiver::HandlePushTask(
           }
         }
       }
+      // If we spilled any return objects to plasma, notify the raylet to pin them.
+      // TODO(edoakes): the plasma objects could be evicted between creating them
+      // here and when raylet pins them.
+      if (!plasma_return_ids.empty()) {
+        RAY_CHECK_OK(
+            local_raylet_client_->PinObjectIDs(caller_address, plasma_return_ids));
+      }
```

Comment:
Are we trying to do distributed reference counting here? Seems like the corresponding unnpin is not yet implemented, so this would end up creating objects that are not deletable? I suggest we postpone this until we have unpin ready, we can discuss more on this.

Also, It should be easier for the core worker to pin the return objects in this case, which is simpler to implement, and can avoid the issues mentioned above.

Reply:
Unpin is implemented, it happens at the raylet when it gets a response to `WaitForObjectEviction`. This happens via a callback in the owner's reference counter when the object goes out of scope or if the worker crashes.

See above for why we're pinning at the raylet.

Reply:
I see, thanks. I somehow misunderstood this part.